video/out/wayland_common.c

```diff
@@ -2070,7 +2084,7 @@ static int set_cursor_visibility(struct vo_wayland_seat *s, bool on)
             struct wl_buffer *buffer = wl_cursor_image_get_buffer(img);
             if (!buffer)
                 return VO_FALSE;
-            int scale = MPMAX(wl->scaling, 1);
+            int scale = MPMAX(wl->scaling_factor, 1);
```

Comment:
I know it was like this before as well, but shouldn't scale be a double?

Reply:
My bad. I missed this was forgotten and it does fix that issue although the inherent problem with libwayland cursor and fractional scaling should still exist (and is fixed by the cursor shape protocol).

Reply:
On second thought, I'm not even sure if we want to put this limit on the scale factor here. It made sense with buffer_scale but maybe it's not needed anymore. Will check later.

Reply:
Yeah this avoids potential truncation to 0 with the cursor. I don't think supporting cursor scales less than 1 has any real practical use, so I think this will stay.